aiida/engine/daemon/execmanager.py

```diff
@@ -75,55 +81,78 @@ def upload_calculation(node, transport, calc_info, folder, inputs=None, dry_run=
                 "[submission of calculation {}] No remote_working_directory configured for computer '{}'".format(
                     node.pk, computer.name))
 
-        # If it already exists, no exception is raised
-        try:
-            transport.chdir(remote_working_directory)
-        except IOError:
-            logger.debug(
-                '[submission of calculation {}] Unable to chdir in {}, trying to create it'.format(
-                    node.pk, remote_working_directory))
+        if calc_info.target_path:
+            pathlist = os.path.normpath(calc_info.target_path).split(os.sep)
+        else:
+            pathlist = [None]
+
+        # If it is an absolute path it will start checking dirs from the root
+        if pathlist[0] == '':
+            transport.chdir('/')
+            pathlist.pop(0)
+
+        # Else (relative path or no target path provided), it will start from
+        # the computer working directory
+        else:
```

Comment:
This should be changed by just defining `remote_working_directory`:
```
if calc_info.target_path:
    if os.path.isabs(calc_info.target_path):
        remote_working_directory = calc_info.target_path
    else:
        remote_working_directory = os.path.join(remote_working_directory, calc_info.target_path)
```

Reply:
Mmm ok, but the existing logic in the execmanager has hardcoded the logic of the 2:2:32 sharding, do you want me to fully replace this for something more versatile? If that is the case, do you think that the "path as list" approach is good? So I mean:

```
(1) set remote_working_directory to the default sharding or to calc_info.target_path
(2) pathlist = os.path.normpath(remote_working_directory).split(os.sep)
(3) Go element by element making sure the dir exists until the last one
(4) Last one should be new or else apply existing procedure of moving this to REMOTE_WORK_DIRECTORY_LOST_FOUND
```

Reply:
Just a comment for the future (as we decided not to implement the `remote_path` yet): the `remote_working_directory` is the base location where the remote directories will be created according to the aiida sharding of the node uuid, not the final path (as the one passed through `target_path`). So these variables may need some re-organization, I can't just replace `remote_working_directory`. Also relevant for this is that, although `remote_working_directory` is almost exclusively used in lines 72 - 93 (where it just checks it exists and chdirs to it), it is also used one single time later (line 113) to create the `LOST_FOUND` directory. I should open an issue with all these things before closing this PR.